streams/to_lines.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Converts a ReadableStream of `Uint8Array` or `string` into an
+ * `AsyncGenerator` of `string`, where each value is divided by a newline at
+ * `\n` or `\r\n`. Trimming the last line if it is empty.
+ *
+ * @param readable A `ReadableStream` of `Uint8Array` or `string`.
+ * @returns An `AsyncGenerator<string>`
+ *
+ * @example JSON Lines
+ * ```ts
+ * import { toLines } from "@std/streams/to-lines";
+ * import { assertEquals } from "@std/assert/assert-equals";
+ *
+ * const readable = ReadableStream.from([
+ *   '{"name": "Alice", "age": ',
+ *   '30}\r\n{"name": "Bob", "age"',
+ *   ": 25}\n",
+ * ]);
+ *
+ * type Person = { name: string, age: number };
+ *
+ * const people: Person[] = []
+ * for await (const line of toLines(readable)) {
+ *   people.push(JSON.parse(line) as Person)
+ * }
+ *
+ * assertEquals(
+ *   people,
+ *   [{ "name": "Alice", "age": 30 }, { "name": "Bob", "age": 25 }],
+ * );
+ * ```
+ */
+export async function* toLines(
+  readable: ReadableStream<Uint8Array> | ReadableStream<string>,
```

Comment:
I think it should just handle `ReadableStream<Uint8Array>`. One could easily just use `readable.pipeThrough(new TextLineStream()` for `ReadableStream<string>`.

Reply:
The original issue suggested it should accept both types of streams, and I think it accepting both would give it more reason to exist

Reply:
The Standard Library already has `TextLineStream` for `ReadableStream<string>`.